app/controllers/budgets_controller.rb

```diff
@@ -0,0 +1,31 @@
+class BudgetsController < ApplicationController
+  def index; end
+
+  def new
+    @budget = current_user.budgets.new
+  end
+
+  def create
+    @budget = current_user.budgets.new(budget_params)
+
+    respond_to do |format|
+      if @budget.save
+        format.html { redirect_to root_path, notice: 'Budget was successfully created.' }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def edit; end
+
+  private
+
+  def set_budget
+    @budget = Budget.find(params[:id])
```

Comment:
I'm pretty sure we want to search only current user's budgets:
```suggestion
    @budget = current_user.find(params[:id])
```